### PR TITLE
Adding runAndCollect

### PR DESCRIPTION
**Pull request by kevmoo**
This is progress towards https://github.com/dart-lang/coverage/issues/36

Just need to add in the entry point the bin

[Review on Reviewable](https://reviewable.io/reviews/dart-lang/coverage/95)

### CODE REVIEW

File: lib/src/util.dart [hunk 1]
```diff
@@ -44,3 +45,22 @@ Future retry(Future f(), Duration interval, {Duration timeout}) async {
...
+
+/// Returns an open port by creating a temporary Socket
+Future<int> getOpenPort() async {
+  ServerSocket socket;
+
+  try {
+    socket = await ServerSocket.bind(InternetAddress.LOOPBACK_IP_V4, 0);
+  } catch (_) {
+    // try again v/ V6 only. Slight possibility that V4 is disabled
+    socket = await ServerSocket.bind(InternetAddress.LOOPBACK_IP_V6, 0,
```
Comment (zoechi) [inline lib/src/util.dart]: Doesn't `bind` try both IPv6 and IPv4 when `v60only` is `false`?

Comment (kevmoo) [inline lib/src/util.dart]: I'm being paranoid. There are still systems that have IPv6 disabled. More likely a system has IPv4 and not v6 then the other way around.

I'm not 100% sure what happens on a system that is v4-only w/ the V6 API.

Comment (zoechi) [inline lib/src/util.dart]: Same here, I hoped you can clarify ;-)

Comment (kevmoo) [inline lib/src/util.dart]: I don't have a machine w/ v6 disabled, so it's tough to figure out. :smile_cat:

**Comment by kevmoo** [comment]
@cbracken I have cookies... :cookie:

**Comment by cbracken** [comment]
I like cookies but I've got a few P0s first. Will get to this by EOW, where EOW is defined as Saturday morning midnight PDT. ;)

**Comment by kevmoo** [comment]
@ekweible @srawlins Could I get a look-through by one of you guys? Look okay?

File: lib/src/collect.dart [hunk 2]
```diff
@@ -42,17 +42,18 @@ Future<Map> _getAllCoverage(VMService service) async {
 
 Future _resumeIsolates(VMService service) async {
   var vm = await service.getVM();
-  var isolateRequests = vm.isolates.map((i) => service.resume(i.id));
-  return Future.wait(isolateRequests);
+  for (var isolate in vm.isolates) {
```
Comment (ekweible) [inline lib/src/collect.dart]: Could we call this `isolateRef`as well to be consistent?

**Comment by ekweible** [comment]
+1 looks good to me.

File: lib/src/collect.dart (hunk 2 above)
Comment (kevmoo) [inline lib/src/collect.dart]: Good point. Fixed

**Pull request merged**